src/_guides/language/effective-dart/design.md

```diff
@@ -952,13 +975,76 @@ class CallChainVisitor {
 }
 {% endprettify %}
 
+This guideline is deliberately softer than the previous one because the benefits
+of static typing are slightly reduced when code is confined within a single
+library. In practice, though, most Dart users type annotate all declarations,
+public or private.
```

Comment:
> In practice, though, most Dart users type annotate all declarations, public or private.

I'm glad to read that (it supports what I've been encouraging my students to do).